### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by RaenonX** [naptha/tesseract.js#853]
**Tesseract.js version (version number for npm/GitHub release, or specific commit for repo)**
v5.0.3

**Describe the bug**
Unable to run. Console showing `maximum call stack size exceeded`.

**To Reproduce**
```typescript
const worker = await createWorker(
  'eng',
  OEM.DEFAULT,
  {
    errorHandler: (error) => {
      console.error('OCR Error', error);
    },
  },
);
await worker.setParameters({
  tessedit_char_blacklist: '$§',
  tessedit_char_whitelist: whitelistChars,
});
```

Images below show what's captured in F12. The `17435` and the regex part is my compiled code, I don't know where the `octet-stream`  from, but it seems to 

![image](https://github.com/naptha/tesseract.js/assets/18443743/eafcc08d-aed8-4956-b52c-ec02b9959de5)
![image](https://github.com/naptha/tesseract.js/assets/18443743/6df932fc-4cb8-48df-ad7a-9aef0ff44ede)

> In the image above that shows the console output, 2 0-height errors show (under 2 `jsdelivr`, the red bar) up immediately after the worker starts. the `octet-stream` one shows up about a minute later.

---------

Image below was the target image to recognize.

![image](https://github.com/naptha/tesseract.js/assets/18443743/0ca2e371-fa8c-4313-9b7f-10ee840e2d51)

**Expected behavior**
No error.

**Device Version:**

This doesn't work:

- OS + Version: SE3 iOS 16.7.1
- Browser: Safari/604.1

This works:

- OS + Version: Windows 11
- Browser: 119.0.2151.58 (Official build) (64-bit)

**Additional context**

**Note**

- I understand that the whole thing might be vague, making it hard to debug, but I've tried what I can do...because my updates in a fast pace, I can't tell exactly when it started breaking.
- This only happens on old iPhone devices it seems. I don't have an old iPhone (this only happens on older iPhones) so it was tested by my friend.
- During my update from v5.0.0 to v5.0.3, I added the option of `tessedit_char_whitelist` and that's it (besides some hidden canvas for storing image data).

**Comment by Balearica** [comment]
I attempted to replicate this issue but was unable to do so.  I used an `iPhone SE 2020 v16.4`, which was the closest device provided by Browserstack to what you describe.  I recognized the provided image on the Tesseract.js [demo site](https://tesseract.projectnaptha.com/), and everything worked as intended.  

I would have your friend try and recognize the image on the Tesseract.js [demo site](https://tesseract.projectnaptha.com/).  This would help determine if the issue is specific to some settings used in your project, or if Tesseract.js consistently encounters issues on that particular device/software combination.
